src/Server/KeeperReadinessHandler.cpp

```diff
@@ -0,0 +1,91 @@
+#include <Server/KeeperReadinessHandler.h>
+
+#if USE_NURAFT
+
+#include <memory>
+
+#include <IO/HTTPCommon.h>
+#include <Coordination/KeeperDispatcher.h>
+#include <Server/HTTPHandlerFactory.h>
+#include <Server/HTTPHandlerRequestFilter.h>
+#include <Server/HTTP/WriteBufferFromHTTPServerResponse.h>
+#include <Poco/Net/HTTPRequestHandlerFactory.h>
+#include <Poco/Net/HTTPServerRequest.h>
+#include <Poco/Net/HTTPServerResponse.h>
+#include <Poco/JSON/JSON.h>
+#include <Poco/JSON/Object.h>
+#include <Poco/JSON/Stringifier.h>
+
+namespace DB
+{
+
+void KeeperReadinessHandler::handleRequest(HTTPServerRequest & /*request*/, HTTPServerResponse & response)
+{
+    try
+    {
+        auto is_leader = keeper_dispatcher->isLeader();
+        auto is_follower = keeper_dispatcher->isFollower() && keeper_dispatcher->hasLeader();
+
+        auto status = is_leader || is_follower;
+
+        Poco::JSON::Object json, details;
+
+        details.set("leader", is_leader);
+        details.set("follower", is_follower);
+        json.set("details", details);
+        json.set("status", status ? "ok": "fail");
+
+        std::ostringstream oss;     // STYLE_CHECK_ALLOW_STD_STRING_STREAM
+        oss.exceptions(std::ios::failbit);
+        Poco::JSON::Stringifier::stringify(json, oss);
```

Comment:
Maybe it would be even simpler to build this JSON manually.... but up to you

Reply:
I'll pass.